infrastructure/comms/src/peer_manager/peer.rs

```diff
@@ -20,18 +20,85 @@
 //  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 //  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-use crate::connection::NetAddress;
+use crate::{connection::net_address::net_addresses::NetAddresses, peer_manager::node_id::NodeId};
+use chrono::prelude::*;
+use tari_crypto::keys::PublicKey;
+use tari_utilities::bit::Bits;
 
+// TODO reputation metric?
+
+#[derive(Debug)]
 pub enum PeerType {
     BaseNode,
     ValidatorNode,
     Wallet,
     TokenWallet,
 }
 
-#[allow(dead_code)]
-pub struct Peer {
-    // TODO: Add fields
-    peer_type: PeerType,
-    addresses: Vec<NetAddress>,
+#[derive(Debug)]
+pub struct Peer<K: PublicKey> {
+    pub public_key: K,
+    pub node_id: NodeId,
+    pub peer_type: PeerType,
```

Comment:
PeerType is definitely a Domain Layer concept, so this should move out of infrastructure.

Reply:
PeerType was moved to domain and a PeerWithType struct was also created.